sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -350,6 +350,7 @@ object ScalaNativePluginInternal {
         val arch = target.split("-").head
         // we need re2 to link the re2 c wrapper (cre2.h)
         val regex = Seq("re2")
+        val zlib  = Seq("z")
```

Comment:
I suggest updating the docs to mention zlib as a required library with instructions on how to install it with header files.